modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,355 @@
+<3> Adds the `etcd` devices with local storage of 10 GB to the cluster.
+====
+
+. Verify that the control plane machines are healthy:
+
+.. Verify the cluster is progressing by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=condition=Progressing=false controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that 3 replica control plane machine sets are running by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.spec.replicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that the 3 control plane machine sets are updated by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.status.updatedReplicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify that the 3 control plane machine sets are healthy by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait --timeout=90m --for=jsonpath='{.status.replicas}'=3 controlplanemachineset.machine.openshift.io -n openshift-machine-api cluster
+----
+
+.. Verify the `ClusterOperators` are progressing in the cluster by running the following command:
+
++
+[source,terminal]
+----
+$ oc wait clusteroperators --timeout=30m --all --for=condition=Progressing=false
+----
+
+. Verify that each of the 3 control plane machines has the additional block device you previously created by running the following script:
+
++
+[source,bash]
+----
+$ cp_machines=$(oc get machines -n openshift-machine-api --selector='machine.openshift.io/cluster-api-machine-role=master' --no-headers -o custom-columns=NAME:.metadata.name) <1>
+
+
+if [[ $(echo "${cp_machines}" | wc -l) -ne 3 ]]; then
+  exit 1
+fi <2>
+
+
+for machine in ${cp_machines}; do
+  if ! oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value.additionalBlockDevices}' | grep -q 'etcd'; then
+	exit 1
+  fi <3>
+done
+----
+<1> Retrieves the machines running in the cluster.
+<2> Checks that there are 3 machines running.
+<3> Checks that 3 `etcd` devices are running on the 3 machines.
+
+. Create a file named `98-var-lib-etcd.yaml` by using the following content:
+
++
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-etcd
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Make File System on /dev/vdb
+          DefaultDependencies=no
+          BindsTo=dev-vdb.device
+          After=dev-vdb.device var.mount
+          Before=systemd-fsck@dev-vdb.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/usr/sbin/mkfs.xfs -f /dev/vdb
```

Comment:
From `man mkfs.xfs`:
>        -f     Force overwrite when an existing filesystem is detected on the device.  By default, mkfs.xfs will not write to the device if it suspects  that
>              there is a filesystem or partition table on the device already.

That doesn't seem right. Isn't that going to blow the filesystem away on second boot? @EmilienM?

Reply:
1. It only happens once
2. The file system is already formatted by Nova, hence the `-f`. Try without and mkfs will fail. That's because the disk comes with `vfat` by default: https://opendev.org/openstack/nova/src/commit/d0b459423dd81644e8d9382b6c87fabaa4f03ad4/nova/privsep/fs.py#L257

Reply:
Is this ok the way that it is or does something have to change?

Reply:
@EmilienM It only happens once as long as it doesn't reboot before you update the configuration to remove it. This is dangerous, and will result in loss of the control plane node if it reboots at the wrong time.

Reply:
In the worst case it could result in complete loss of the control plane, e.g. a datacenter power outage.

Reply:
This procedure needs to happen on day 1, right after deploying a control plane, btw.
Also this is a TechPreview procedure, this is not officially supported yet and I believe what you're rightfully asking is that we rework that procedure before we make it GA.

While the current procedure is terrible, this is something deliverable as TechPreview IMO, by explaining that:
1. Moving etcd needs to happen RIGHT AFTER deploying the control plane, before any workload is running
2. The procedure is TechPreview and will be improved & secured in the future so customers will be able to move etcd safely on day 2.

Reply:
@mdbooth it sounds like we need a warning in two spots. Give me a second, and I'll push an update.